From 5596be8dfacbe51b5da9232636af3a3c7fdaf458 Mon Sep 17 00:00:00 2001
From: Haiyang Zhang <haiyangz@microsoft.com>
Date: Fri, 1 Sep 2017 14:30:04 -0700
Subject: [PATCH 90/98] hv_netvsc: Clean up an unused parameter in
 rndis_filter_set_rss_param()

This patch removes the parameter, num_queue in
rndis_filter_set_rss_param(), which is no longer in use.

Signed-off-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/hyperv/hyperv_net.h   |    2 +-
 drivers/net/hyperv/netvsc_drv.c   |    2 +-
 drivers/net/hyperv/rndis_filter.c |    5 ++---
 3 files changed, 4 insertions(+), 5 deletions(-)

--- a/drivers/net/hyperv/hyperv_net.h	2017-10-05 14:20:55.947579162 -0700
+++ b/drivers/net/hyperv/hyperv_net.h	2017-10-05 14:20:55.947579162 -0700
@@ -213,7 +213,7 @@ void rndis_filter_update(struct netvsc_d
 void rndis_filter_device_remove(struct hv_device *dev,
 				struct netvsc_device *nvdev);
 int rndis_filter_set_rss_param(struct rndis_device *rdev,
-			       const u8 *key, int num_queue);
+			       const u8 *key);
 int rndis_filter_receive(struct net_device *ndev,
 			 struct netvsc_device *net_dev,
 			 struct hv_device *dev,
--- a/drivers/net/hyperv/netvsc_drv.c	2017-10-05 14:20:55.947579162 -0700
+++ b/drivers/net/hyperv/netvsc_drv.c	2017-10-05 14:20:55.947579162 -0700
@@ -1424,7 +1424,7 @@ static int netvsc_set_rxfh(struct net_de
 		key = rndis_dev->rss_key;
 	}
 
-	return rndis_filter_set_rss_param(rndis_dev, key, ndev->num_chn);
+	return rndis_filter_set_rss_param(rndis_dev, key);
 }
 
 /* Hyper-V RNDIS protocol does not have ring in the HW sense.
--- a/drivers/net/hyperv/rndis_filter.c	2017-10-05 14:20:55.947579162 -0700
+++ b/drivers/net/hyperv/rndis_filter.c	2017-10-05 14:20:55.947579162 -0700
@@ -717,7 +717,7 @@ cleanup:
 }
 
 int rndis_filter_set_rss_param(struct rndis_device *rdev,
-			       const u8 *rss_key, int num_queue)
+			       const u8 *rss_key)
 {
 	struct net_device *ndev = rdev->ndev;
 	struct rndis_request *request;
@@ -1258,8 +1258,7 @@ struct netvsc_device *rndis_filter_devic
 		   atomic_read(&net_device->open_chn) == net_device->num_chn);
 
 	/* ignore failues from setting rss parameters, still have channels */
-	rndis_filter_set_rss_param(rndis_device, netvsc_hash_key,
-				   net_device->num_chn);
+	rndis_filter_set_rss_param(rndis_device, netvsc_hash_key);
 out:
 	if (ret) {
 		net_device->max_chn = 1;
